File: src/main/java/com/dashboard/interfaces/controllerInterfaces/EmployeeControllerInterface.java
package com.dashboard.interfaces.controllerInterfaces;

import com.dashboard.models.JPA.Employee;
import io.swagger.v3.oas.annotations.media.Content;
import io.swagger.v3.oas.annotations.media.Schema;
import io.swagger.v3.oas.annotations.responses.ApiResponse;
import io.swagger.v3.oas.annotations.responses.ApiResponses;
import io.swagger.v3.oas.annotations.tags.Tag;
import org.springframework.data.domain.Pageable;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@Tag(name = "Employees")
public interface EmployeeControllerInterface {

    @PostMapping
    @ApiResponses(value = {@ApiResponse(description = "Successfully created new employee",
            responseCode = "201",
            content = @Content(schema = @Schema(implementation = ResponseEntity.class)))})
    ResponseEntity<?> createEmployee(@RequestBody Employee employee);

    @GetMapping
    @ApiResponses(value = {@ApiResponse(description = "Get all employees",
            responseCode = "200",
            content = @Content(schema = @Schema(implementation = ResponseEntity.class)))})
    ResponseEntity<?> getAllEmployees(Pageable pageable);

    @GetMapping("/{id}")
    @ApiResponses(value = {@ApiResponse(description = "Get a specific employee with specific ID",
            responseCode = "200",
            content = @Content(schema = @Schema(implementation = ResponseEntity.class)))})
    ResponseEntity<?> getEmployeeById(@PathVariable Integer id);

    @DeleteMapping("/{id}")
    @ApiResponses(value = {@ApiResponse(description = "Delete a employee with specific ID",
            responseCode = "200",
            content = @Content(schema = @Schema(implementation = ResponseEntity.class)))})
    ResponseEntity<?> deleteEmployeeById(@PathVariable Integer id);

    @PutMapping("/{id}")
    @ApiResponses(value = {@ApiResponse(description = "Update an employee with specific ID",
            responseCode = "200",
            content = @Content(schema = @Schema(implementation = ResponseEntity.class)))})
    ResponseEntity<?> updateEmployee(@PathVariable Integer employeeId, @RequestBody Employee employee);

    // Task Operation with Employee
    @PostMapping("/{employeeID}/tasks/{taskID}")
    @ApiResponses(value = {@ApiResponse(description = "Update a employees with specific ID",
            responseCode = "200",
            content = @Content(schema = @Schema(implementation = ResponseEntity.class)))})
    ResponseEntity<?> assignTask(@PathVariable Integer employeeID, @PathVariable Integer taskID);

    @DeleteMapping("/{employeeID}/tasks/{taskID}")
    @ApiResponses(value = {@ApiResponse(description = "Update a employees with specific ID",
            responseCode = "200",
            content = @Content(schema = @Schema(implementation = ResponseEntity.class)))})
    ResponseEntity<?> unassignTask(@PathVariable Integer employeeID, @PathVariable Integer taskID);
}

File: src/main/kotlin/dashboard/controller/interfaces/EmployeeControllerInterface.kt
package dashboard.controller.interfaces

import dashboard.model.Employee
import org.springframework.data.domain.Pageable
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

interface EmployeeControllerInterface {
    
    @PostMapping
    fun createEmployee(@RequestBody employee: Employee): ResponseEntity<*>

    @GetMapping
    fun getAllEmployees(): ResponseEntity<*>

    @GetMapping("/{id}")
    fun getEmployeeById(@PathVariable id: Int): ResponseEntity<*>

    @DeleteMapping("/{id}")
    fun deleteEmployeeById(@PathVariable id: Int): ResponseEntity<*>

    @PutMapping("/{id}")
    fun updateEmployee(@PathVariable employeeId: Int, @RequestBody employee: Employee): ResponseEntity<*>

    // Task Operation with Employee
    @PostMapping("/{employeeID}/tasks/{taskID}")
    fun assignTask(@PathVariable employeeID: Int, @PathVariable taskID: Int): ResponseEntity<*>

    @DeleteMapping("/{employeeID}/tasks/{taskID}")
    fun unassignTask(@PathVariable employeeID: Int, @PathVariable taskID: Int): ResponseEntity<*>
}